autotest/utilities/test_gdalinfo.py

```diff
@@ -1109,3 +1109,44 @@ def test_gdalinfo_if_option():
         encoding="UTF-8",
     )
     assert err is not None
+
+
+###############################################################################
+# Test STAC JSON output
+
+
+def test_gdalinfo_stac_json():
+    if test_cli_utilities.get_gdalinfo_path() is None:
+        pytest.skip()
+
+    ret, _ = gdaltest.runexternal_out_and_err(
+        test_cli_utilities.get_gdalinfo_path()
+        + " -json -proj4 -stats -hist ../gcore/data/byte.tif",
+        encoding="UTF-8",
+    )
```

Comment:
computing statistics directly on reference files has side effects on other tests, so:
```suggestion
    tmpfilename = "tmp/test_gdalinfo_stac_json.tif"
    shutil.copy("../gcore/data/byte.tif", tmpfilename)
    ret, _ = gdaltest.runexternal_out_and_err(
        test_cli_utilities.get_gdalinfo_path()
        + " -json -proj4 -stats -hist " + tmpfilename,
        encoding="UTF-8",
    )
    gdal.GetDriverByName("GTiff").Delete(tmpfilename)
```